cmd/eksctl-anywhere/cmd/generate_tinkerbelltemplateconfig.go

```diff
@@ -0,0 +1,91 @@
+		BootstrapTinkerbellIP string
+	}
+
+	// Configure the flagset. Some of these flags are duplicated from other parts of the cmd code
+	// for consistency but their descriptions may vary because of the commands use-case.
+	flgs := pflag.NewFlagSet("tinkerbelltemplateconfig", pflag.ExitOnError)
+	flgs.StringVarP(&opts.fileName, "filename", "f", "",
+		"A path to the EKS Anywhere cluster specification")
+	flgs.StringVar(&opts.bundlesOverride, "bundles-override", "",
+		"A path to a custom bundles manifest")
+	flgs.StringVar(&opts.BootstrapTinkerbellIP, "tinkerbell-bootstrap-ip", "",
+		"The IP used to expose the Tinkerbell stack from the bootstrap cluster")
+
+	cmd := &cobra.Command{
+		Use:     "tinkerbelltemplateconfig",
+		Aliases: []string{"ttc"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Validation logic called by newClusterSpec arbitrarily logs warnings. Until it can
+			// be refactored we need to redirect logging such that the output is swallowed.
+			if err := logger.Init(logger.Options{
+				Level: -1,
+			}); err != nil {
+				return err
+			}
+
+			cs, err := newClusterSpec(opts.clusterOptions)
+
+			// Reset the logger before evaluating anything in-case logic higher up the call path
+			// needs the logger.
+			if err := logger.Init(logger.Options{
+				Level: viper.GetInt("verbosity"),
+			}); err != nil {
+				return err
+			}
+
+			// Handle the newClusterSpec() error.
+			if err != nil {
+				return err
+			}
+
+			// Generating the TinkerbellTemplateConfig requires the OS family to ensure the right
+			// actions are produced. The OS family is specified per TinkerbellMachineConfig,
+			// therefore is specified in multiple places (control plane and worker node groups).
+			// However, we only support single OS family clusters so validation should error out
+			// earlier if they aren't the same. This means we can use the control plane machine
+			// configs OS family.
+			controlPlaneMachineConfigName := cs.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
+			controlPlaneMachineConfig := cs.TinkerbellMachineConfigs[controlPlaneMachineConfigName]
+			osFamily := controlPlaneMachineConfig.OSFamily()
+
+			// For modular upgrades the version bundle is retrieve per worker node group. However,
+			// because Tinkerbell action images are the same for every Kubernetes version within
+			// the same bundle manifest, its OK to just use the root version bundle.
+			bundle := *cs.RootVersionsBundle().VersionsBundle
+
+			osImageURL := cs.TinkerbellDatacenter.Spec.OSImageURL
+			tinkerbellIP := cs.TinkerbellDatacenter.Spec.TinkerbellIP
+
+			cfg := v1alpha1.NewDefaultTinkerbellTemplateConfigCreate(cs.Cluster, bundle, osImageURL,
+				opts.BootstrapTinkerbellIP, tinkerbellIP, osFamily)
+
+			return yaml.NewK8sEncoder(os.Stdout).Encode(cfg)
+		},
+	}
+
+	// Configure the commands flags.
+	cmd.Flags().AddFlagSet(flgs)
+	flags.MarkRequired(cmd.Flags(), []string{"filename", "tinkerbell-bootstrap-ip"})
```

Comment:
Note the bootstrap IP is required. The IP is generally whatever is assigned to the admin machine. We auto detect it in `create cluster` but that didn't feel like a good idea here given the IP could change between this `generate` and `create` being invoked.